pyfr/libs/conduit.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+
+from ctypes import (POINTER, c_char_p, c_void_p, 
+                    c_double, c_float, c_long, c_longlong, RTLD_GLOBAL)
+from pyfr.ctypesutil import LibWrapper
+
+
+class ConduitWrappers(LibWrapper):
+    _libname = 'conduit'
+    _mode = RTLD_GLOBAL
+
+    # Functions
+    _functions = [
+        (c_void_p, 'conduit_node_append', c_void_p),
+        (c_void_p, 'conduit_node_create', c_void_p),
+        (None, 'conduit_node_destroy', c_void_p),
+        (c_void_p, 'conduit_node_fetch', c_void_p, c_char_p),
+        (c_char_p, 'conduit_node_fetch_path_as_char8_str', c_void_p, 
+         c_char_p),
+        (POINTER(c_float), 'conduit_node_fetch_path_as_float32_ptr', c_void_p,
+         c_char_p),
+        (POINTER(c_double), 'conduit_node_fetch_path_as_float64_ptr', 
+         c_void_p, c_char_p),
+        (POINTER(c_longlong), 'conduit_node_fetch_path_as_int64_ptr', 
+         c_void_p, c_char_p),
+        (c_longlong, 'conduit_node_total_bytes_allocated', c_void_p),
+        (c_void_p, 'conduit_node_set_path_char8_str', c_void_p, c_char_p,
+         c_char_p),
+        (c_void_p, 'conduit_node_set_path_float32', c_void_p, c_char_p,
+         c_float),
+        (c_void_p, 'conduit_node_set_path_float32_ptr', c_void_p, c_char_p,
+         POINTER(c_double), c_longlong),
+        (c_void_p, 'conduit_node_set_path_float64', c_void_p, c_char_p, 
+         c_double),
+        (c_void_p, 'conduit_node_set_path_float64_ptr', c_void_p, c_char_p,
+         POINTER(c_double), c_longlong),
+        (c_void_p, 'conduit_node_set_path_int32', c_void_p, c_char_p, c_long),
+        (c_void_p, 'conduit_node_set_path_int64', c_void_p, c_char_p,
+         c_longlong),
+        (c_void_p, 'conduit_node_set_path_int64_ptr', c_void_p, c_char_p,
+         POINTER(c_longlong), c_longlong),
+        (c_void_p, 'conduit_node_set_path_node', c_void_p, c_char_p, c_void_p), # maybe this is copying, external might be better
+        (c_void_p, 'conduit_node_print', c_void_p),
+        (c_void_p, 'conduit_node_print_detailed', c_void_p),
+    ]
+
+
+class ConduitNode:
+    def __init__(self, ptr=None, child=False):
+        self.lib = ConduitWrappers()
+        self.child = child
+
+        if ptr is not None:
+            self.ptr = ptr
+        else:
+            self.ptr = c_void_p(self.lib.conduit_node_create(None))
+
+    def __del__(self):
+        if hasattr(self, 'lib') and self.ptr is not None and not self.child:
+            self.lib.conduit_node_destroy(self.ptr)
+        self.ptr = None
+
+    def __getitem__(self, key):
```

Comment:
Is this needed?

Reply:
You asked me to add it for safety

Reply:
We should try to remove it.  The semantics of these things are a royal pain to get right (as you need to be sure that the you fetch node will not be destroyed by someone else).